tests/pipeline/test_pipeline.py

```diff
@@ -235,6 +235,50 @@ def test_with_custom_config(self, snapshot: SnapshotTest):
 
         snapshot.assert_match(enriched_pipeline, "test-pipeline")
 
+    def test_with_custom_config_with_absolute_defaults_path(
+        self, snapshot: SnapshotTest
+    ):
+        with open(RESOURCE_PATH / "custom-config/config.yaml", "r") as rel_config_yaml:
+            config_dict = yaml.safe_load(rel_config_yaml)
+        config_dict["defaults_path"] = str(
+            (RESOURCE_PATH / "no-topics-defaults").absolute()
+        )
+        temp_config_path = RESOURCE_PATH / "custom-config/temp_config.yaml"
+        try:
+            with open(temp_config_path, "w") as abs_config_yaml:
+                yaml.dump(config_dict, abs_config_yaml)
+                result = runner.invoke(
```

Comment:
dedent this line and below. only `yaml.dump` needs to be inside the context manager, i.e. writing to the file. afterwards the `abs_config_yaml` file can be closed